vyperlang/vyper
# Updated ERC1155 ownable to support opensea collection metadata

### What I did
Updated the ERC1155 ownable example to fully support opensea.io

### How I did it
During online testing I ran into 2 missing functions that are needed to correctly display the ERC1155 contract's metadata on opensea. setContractURI and contractURI have been added.
[edit] The tests have also been updated [/edit]
```# URI #
@external
def setContractURI(contractUri: String[MAX_URI_LENGTH]):
    """
    @dev set the contractURI for the contract. points to collection metadata file
    @param contractUri the new urcontractUri for the contract
    """
    assert not self.paused, "The contract has been paused"
    assert self._contractUri != contractUri, "new and current URI are identical"
    assert msg.sender == self.owner, "Only the contract owner can update the URI"
    self._contractUri = contractUri
    log evtContractURI(contractUri, 0)


@view
@external
def contractURI() -> String[MAX_URI_LENGTH]:
    """
    @dev retrieve the contractUri to retrieve the collection metadata
    """
    return self._contractUri
```

### How to verify it
Deploy on a testnet, then add the collection on opensea.io 
1. go to https://opensea.io/get-listed 
2. click "live on a testnet"
3. enter which testnet + contract address, click submit

The contract should show up with metadata


### Commit message

Commit message for the final, squashed PR. (Optional, but reviewers will appreciate it! Please see [our commit message style guide](../../blob/master/docs/style-guide.rst#best-practices-1) for what we would ideally like to see in a commit message.)

### Description for the changelog

### Cute Animal Picture
![image](https://user-images.githubusercontent.com/33023023/170331376-6b09ac4f-e568-490b-beaf-28ff99bf93bb.png)
